routing/control_tower.go

```diff
@@ -280,7 +280,7 @@ func (p *controlTower) SubscribePayment(paymentHash lntypes.Hash) (
 	// updates. Otherwise this update is the final update and the incoming
 	// channel can be closed. This will close the queue's outgoing channel
 	// when all updates have been written.
-	if payment.Status == channeldb.StatusInFlight {
+	if !payment.Terminated() {
```

Comment:
Maybe this commit and the previous one can be squashed with the introduction of `updatable` to prevent flipping back and forth between commits during review.

Reply:
done!